src/nimibook/defaults.nim

```diff
@@ -31,7 +31,11 @@ proc useNimibook*(nbDoc: var NbDoc) =
   nbDoc.context["git_repository_url"] = "https://github.com/pietroppeter/nimibook"
   nbDoc.context["git_repository_icon"] = "fa-github"
 
-  var toc = load("../book/toc.json")
+  let tocPath = "../book/toc.json"
+  if not tocPath.fileExists:
+    withDir("..".AbsoluteDir): # todo: add a withDir for RelativeDir in nimib /paths
```

Comment:
"..".AbsoluteDir 

Isn't ".." a relative dir and not an absolute dir ?

Could that be replaced by ``getProjectPath.parentDir()`` (or maybe ``currentSourcePath.parentDir()``) or will that pose problem when distributing the binary ?

Reply:
yes, it is, see the comment. I like to use `withDir` which I added in nimib /paths; it came from nimscript and was added in fusion which I copied and pasted into nimib (since I was not sure how to put fusion requirements in nimble file). I want to add also a `withDir` with relative directory, but until now it is not there, so now it looks bad (at least it is good that you were able to notice it).

Regarding the other proposals, the rationale is that, nimib will currently use as homeDir `docs` and it would by default change currentDir to homeDir so a `..`.RelativeDir should always work with thefolder structure we plan `nimibook` to have. For your other suggestions maybe `getProjectPath` could work (does it get the dir of nimble file?), `currentSourcePath` probably not (since source could be in subfolders of `book`).

Anyway I will need to revise how paths are handled in nimib, since nimibook has challenged the choices I made, so the idea is: 1) for now have workarounds in nimibook; 2) understand a better design and implement in nimib; 3) update nimibook with the improved design from nimib